huaweicloud/services/codeartsdeploy/resource_huaweicloud_codearts_deploy_application_group.go

```diff
@@ -0,0 +1,318 @@
+			},
+			"created_by": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the ID of the group creator.`,
+			},
+			"updated_by": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the ID of the user who last updates the group.`,
+			},
+			"application_count": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `Indicates the total number of applications in the group.`,
+			},
+			"children": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: `Indicates the child group name list.`,
+			},
+		},
+	}
+}
+
+func resourceDeployApplicationGroupCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.NewServiceClient("codearts_deploy", region)
+	if err != nil {
+		return diag.Errorf("error creating CodeArts deploy client: %s", err)
+	}
+
+	createHttpUrl := "v1/projects/{project_id}/applications/groups"
+	createPath := client.Endpoint + createHttpUrl
+	createPath = strings.ReplaceAll(createPath, "{project_id}", d.Get("project_id").(string))
+	createOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json;charset=utf-8",
+		},
+		JSONBody: utils.RemoveNil(buildCreateDeployApplicationGroupBodyParams(d)),
+	}
+
+	_, err = client.Request("POST", createPath, &createOpt)
+	if err != nil {
+		return diag.Errorf("error creating application group: %s", err)
+	}
+
+	// get groups list
+	groups, err := getDeployApplicationGroup(client, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// filter by its name and parent ID
+	groupId := filterGroupByNameAndParentId(groups.([]interface{}), d)
+	if groupId == "" {
+		return diag.Errorf("unable to find group ID from API response")
+	}
+
+	d.SetId(groupId)
+
+	return resourceDeployApplicationGroupRead(ctx, d, meta)
+}
+
+func buildCreateDeployApplicationGroupBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"name":      d.Get("name"),
+		"parent_id": utils.ValueIgnoreEmpty(d.Get("parent_id")),
+	}
+
+	return bodyParams
+}
+
+func resourceDeployApplicationGroupRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.NewServiceClient("codearts_deploy", region)
+	if err != nil {
+		return diag.Errorf("error creating CodeArts deploy client: %s", err)
+	}
+
+	// get groups list
+	groups, err := getDeployApplicationGroup(client, d)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving application groups")
+	}
+
+	// filter group by group ID
+	group := filterGroupByGroupId(groups.([]interface{}), d.Id())
+	if group == nil {
```

Comment:
```suggestion
	if group == "" {
```
the default value is empt, not nil